shacl12-ui/index.html

```diff
@@ -801,8 +801,592 @@ <h3>Document Conventions</h3>
+	...</pre>
+        </section>
+        <section id="TextAreaEditor">
+          <h4>shui:TextAreaEditor</h4>
+          <p>
+            <b>Score:</b>
+            0 if the property is marked <code>sh:singleLine true</code>.
+            20 if the value is an <code>xsd:string</code> literal and <code>sh:singleLine false</code>.
+            5 if the value is an <code>xsd:string</code> literal.
+            2 if the property has <code>xsd:string</code> among the permissible datatypes.
+            <code>null</code> if the property has a custom datatype (not from xsd or rdf namespaces but for example <code>geo:wktLiteral</code>).
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a multi-line text area to enter the value of a literal.
+          </p>
+          <img src="images/editors/TextAreaEditor.png" />
+          <pre class="example-shapes">
+ex:Country-description
+	a sh:PropertyShape ;
+	sh:path ex:description ;
+	<b>sh:datatype xsd:string ;
+	sh:singleLine false ;</b>
+	...</pre>
+        </section>
+        <section id="TextAreaWithLangEditor">
+          <h4>shui:TextAreaWithLangEditor</h4>
+          <p>
+            <b>Score:</b>
+            0 if the property is marked <code>sh:singleLine true</code>.
+            15 if the value is an <code>rdf:langString</code> literal and <code>sh:singleLine false</code>.
+            5 if the value is an <code>rdf:langString</code> literal or the property permits such values.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a multi-line text area to enter the value of a literal and a drop-down to select a language.
+          </p>
+          <img src="images/editors/TextAreaWithLangEditor.png" />
+          <pre class="example-shapes">
+ex:Country-description
+	a sh:PropertyShape ;
+	sh:path ex:description ;
+	<b>sh:datatype rdf:langString ;
+	sh:singleLine false ;</b>
+	...</pre>
+        </section>
+        <section id="TextFieldEditor">
+          <h4>shui:TextFieldEditor</h4>
+          <p>
+            <b>Score:</b>
+            10 if the value is a literal that is neither <code>rdf:langString</code> nor <code>xsd:boolean</code>.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            an input field to enter the value of a literal, without the ability to change language or datatype.
+          </p>
+          <img src="images/editors/TextFieldEditor.png" />
+          <pre class="example-shapes">
+ex:Country-code
+	a sh:PropertyShape ;
+	sh:path ex:code ;
+	<b>sh:datatype xsd:string ;</b>
+	...</pre>
+        </section>
+        <section id="TextFieldWithLangEditor">
+          <h4>shui:TextFieldWithLangEditor</h4>
+          <p>
+            <b>Score:</b>
+            11 if the value is an <code>rdf:langString</code> literal or the property permits either (both) <code>rdf:langString</code> or <code>xsd:string</code>.
```

Comment:
Permitting `either` means it permits _one_ datatype at a time, which I think is correct. Permitting `both` means it permits *two* datatypes at the same time, which I think is incorrect, as I think these are disjoint types.
```suggestion
            11 if the value is an <code>rdf:langString</code> literal or the property permits either <code>rdf:langString</code> or <code>xsd:string</code>.
```